backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java

```diff
@@ -381,4 +391,96 @@ public FutureExperimentDAO(
               executor);
     }
   }
+
+  public InternalFuture<Experiment> updateExperimentNameOrDescription(
+      UpdateExperimentNameOrDescription request) {
+    return getProjectIdByExperimentId(Collections.singletonList(request.getId()))
+        .thenCompose(
+            projectIdFromExperimentMap ->
+                futureProjectDAO.checkProjectPermission(
+                    projectIdFromExperimentMap.get(request.getId()), ModelDBServiceActions.UPDATE),
+            executor)
+        .thenCompose(
+            unused -> {
+              var name = request.getName();
+              if (request.getName().isEmpty()) {
+                name = MetadataServiceImpl.createRandomName();
+              }
```

Comment:
@conradoverta 
 When updating a name do we want to pick a random name if a blank one is passed?   I can see this behavior when creating a new experiment, but it seems an odd behavior when updating.